cluster-http/src/main/scala/akka/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -33,6 +37,8 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
     system.dynamicAccess.createInstanceFor[ServiceDiscovery](clazz, List(classOf[ActorSystem] → system)).get
   }
 
+  @volatile private[this] var _selfContactPointUri: Uri = _
```

Comment:
would it 'ripple through' too much to make this a promise?

Reply:
Hmm, that could work :)